src/Player/Player.h

```diff
@@ -67,7 +67,12 @@ class Player
   /// issueOrder
   /// Take in an order and add it into the OrderList
   /// </summary>
-  vector<Order*> issueOrder(Order* o);
+  OrdersList* issueOrder(Order* o);
+  /**
+   * @brief Set the Orders List Null to remove dangling pointer
+   * 
+   */
+  void setOrdersListNull();
```

Comment:
as explained in the playerdriver comment, the players destructor should remove the orders list. so instead of having this method, I called `delete p` which will trigger `~Player` that looks like this:

<img width="531" alt="image" src="https://github.com/LaraMo/Risk-Swift-Sloths/assets/31352301/a150b37d-d685-4822-888e-2d5c6bb2fbda">

The changes are in the player MR. you can leave ur MR like this.

Reply:
Oh i get it, I did this before the MR so i didn't see it!